src/main/java/org/apache/rocketmq/flink/source/reader/RocketMQPartitionSplitReader.java

```diff
@@ -173,25 +177,28 @@ public RecordsWithSplitIds<Tuple3<T, Long, Long>> fetch() throws IOException {
                         recordsBySplits.prepareForRead();
                         return recordsBySplits;
                     }
-                    if (StringUtils.isNotEmpty(sql)) {
-                        pullResult =
-                                consumer.pull(
-                                        messageQueue,
-                                        MessageSelector.bySql(sql),
-                                        messageOffset,
-                                        MAX_MESSAGE_NUMBER_PER_BLOCK);
-                    } else {
-                        pullResult =
-                                consumer.pull(
-                                        messageQueue,
-                                        tag,
-                                        messageOffset,
-                                        MAX_MESSAGE_NUMBER_PER_BLOCK);
-                    }
-                } catch (MQClientException
-                        | RemotingException
-                        | MQBrokerException
-                        | InterruptedException e) {
+                    consumer.assign(Collections.singletonList(messageQueue));
```

Comment:
The same issue. It would be better if we handle the subscription and assignment asynchronously rather than in a `fetch` call.

Reply:
actually, seek method is enough，i will remove assign code